src/layman/common/prime_db_schema/publications.py

```diff
@@ -76,24 +80,24 @@ def get_publication_infos_with_metainfo(workspace_name=None, pub_type=None, styl
             -- A∩B / (A + B)
             CASE
                 -- if there is any intersection
-                WHEN p.bbox && ST_MakeBox2D(ST_MakePoint(%s, %s),
-                                            ST_MakePoint(%s, %s))
+                WHEN ST_TRANSFORM(ST_SetSRID(p.bbox, p.srid), %s) && ST_SetSRID(ST_MakeBox2D(ST_MakePoint(%s, %s),
+                                                                                             ST_MakePoint(%s, %s)), %s)
                     THEN
                         -- in cases, when area of intersection is 0, we want it rank higher than no intersection
-                        GREATEST(st_area(st_intersection(p.bbox, ST_MakeBox2D(ST_MakePoint(%s, %s),
-                                                                              ST_MakePoint(%s, %s)))),
+                        GREATEST(st_area(st_intersection(ST_TRANSFORM(ST_SetSRID(p.bbox, p.srid), %s), ST_SetSRID(ST_MakeBox2D(ST_MakePoint(%s, %s),
+                                                                                                                               ST_MakePoint(%s, %s)), %s))),
                                  1)
                         -- we have to solve division by 0
-                        / (GREATEST(st_area(p.bbox), 1) +
-                           GREATEST(st_area(ST_MakeBox2D(ST_MakePoint(%s, %s),
-                                                         ST_MakePoint(%s, %s))),
+                        / (GREATEST(st_area(ST_TRANSFORM(ST_SetSRID(p.bbox, p.srid), %s)), 1) +
+                           GREATEST(st_area(ST_SetSRID(ST_MakeBox2D(ST_MakePoint(%s, %s),
+                                                                    ST_MakePoint(%s, %s)), %s)),
                                     1)
```

Comment:
I think we should make this 1 much smaller because of EPSG:4326. I suggest 0.00001, it is about one meter in ESPG:4326.